### PR TITLE
Improved dotnet project info to be closer to how the dnx project info…

### PR DESCRIPTION
**Pull request by david-driscoll**
… used to be

### CODE REVIEW

**Comment by david-driscoll** [comment]
cc @DustinCampbell @troydai 

Basically things brings back some of the fields we had before with `DnxProject`, list of configurations / frameworks, etc.

File: src/OmniSharp.DotNet/DotNetProjectSystem.cs [hunk 1]
```diff
@@ -181,10 +163,30 @@ private void UpdateProject(string projectDirectory)
...
+        }
+
+        private void AddProject(ProjectId id, ProjectContext context)
+        {
+            var info = ProjectInfo.Create(
+                id: id,
+                version: VersionStamp.Create(),
+                name: $"{context.ProjectFile.Name}+{context.TargetFramework.GetShortFolderName()}",
+                assemblyName: context.ProjectFile.Name,
+                language: LanguageNames.CSharp,
```
Comment (filipw) [inline src/OmniSharp.DotNet/DotNetProjectSystem.cs]: respect the `Language` property of the `IProjectSystem` :)

**Comment by DustinCampbell** [comment]
The change seems righteous, though I'll need to make a significant change on the VS Code side to absorb this change. :smile:

I'm not all that familiar with how configurations are represented in project.json. I assume that VS Code will need some way to switch configurations and track the "active" configuration?

**Comment by david-driscoll** [comment]
Yeah, in atom we have support with semantic highlight to fade out `excluded code`, if you have a specific framework selected.  We haven't done much with configuration yet, but I'm sure it's something in the future we'll need.

File: src/OmniSharp.DotNet/Cache/ProjectStatesCache.cs [hunk 2]
```diff
@@ -5,21 +5,73 @@
 using Microsoft.DotNet.ProjectModel;
 using Microsoft.Extensions.Logging;
 using NuGet.Frameworks;
+using OmniSharp.DotNet.Models;
+using OmniSharp.Models;
+using OmniSharp.Services;
 
 namespace OmniSharp.DotNet.Cache
 {
+    public class ProjectEntry
```
Comment (troydai) [inline src/OmniSharp.DotNet/Cache/ProjectStatesCache.cs]: Move it to its own file?

File: src/OmniSharp.DotNet/Cache/ProjectStatesCache.cs [hunk 3]
```diff
@@ -62,6 +115,15 @@ public IReadOnlyCollection<ProjectState> GetValues()
                     addAction(projectId, context);
                 }
             }
+
+            var projectInformation = new DotNetProjectInformation(entry);
+            if (added)
+            {
+                EmitProject(EventTypes.ProjectChanged, projectInformation);
+            }
+            else {
```
Comment (troydai) [inline src/OmniSharp.DotNet/Cache/ProjectStatesCache.cs]: Move the open brace to its own line.

File: src/OmniSharp.DotNet/Cache/ProjectStatesCache.cs [hunk 4]
```diff
@@ -128,49 +190,34 @@ private ProjectEntry GetOrAddEntry(string projectDirectory)
...
-                return $"ProjectEntry {ProjectDirectory}, {_states.Count} states";
-            }
+        private void EmitProject(string eventType, DotNetProjectInformation information) {
+            _emitter.Emit(
+                eventType,
+                new ProjectInformationResponse()
+                {
+                    { "DotNetProject", information },
+                    // the key is hard coded in VSCode
+                    { "DnxProject", information }
```
Comment (troydai) [inline src/OmniSharp.DotNet/Cache/ProjectStatesCache.cs]: @DustinCampbell can we fix this in VSCode?

File: src/OmniSharp.DotNet/DotNetProjectSystem.cs [hunk 5]
```diff
@@ -119,10 +114,13 @@ public void Update(bool allowRestore)
...
 
-            _projectStates.RemoveExcept(projectPaths, id =>
+            _projectStates.RemoveExcept(projectPaths, entry =>
             {
-                _omnisharpWorkspace.RemoveProject(id);
-                _logger.LogInformation($"Removing project {id.Id}.");
+                foreach (var state in entry.ProjectStates)
+                {
+                    _omnisharpWorkspace.RemoveProject(state.Id);
+                    _logger.LogInformation($"Removing project {state.Id}.");
```
Comment (troydai) [inline src/OmniSharp.DotNet/DotNetProjectSystem.cs]: I was thinking downgrade the log level for this to _Debug_

**Comment by troydai** [comment]
:shipit:

File: src/OmniSharp.DotNet/Cache/ProjectStatesCache.cs (hunk 4 above)
Comment (DustinCampbell) [inline src/OmniSharp.DotNet/Cache/ProjectStatesCache.cs]: I'm not sure why it's necessary at this point. VS Code has DnxProject while we're targeting an older OmniSharp, but I'm not sure why we need this here.

Comment (david-driscoll) [inline src/OmniSharp.DotNet/Cache/ProjectStatesCache.cs]: I just left it in, wasn't sure if was okay to remove.  I have no issues taking it out.

**Pull request merged**